### PR TITLE
Tracking accordions

### PR DESCRIPTION
**Pull request by johnsibly**
@acnalesso|@frikille:
Below is what we have done in this PR:
* Changed accordions to anchors and added data-tracking-context="opening|closing"
* Fixed AccordionItem props issue

Cheers

### CODE REVIEW

**Comment by peter-mouland** [comment]
who is we?

**Comment by johnsibly** [comment]
I believe we left our initials in the title? No this is not WIP.

**Comment by johnsibly** [comment]
we are @acnalesso @frikille

File: src/toolkit.sass [hunk 1]
```diff
@@ -13,3 +13,11 @@
...
 @import './node_modules/nowtv-toolkit/styles/components/button'
 @import './fonts'
+
+.no-select
+  -webkit-touch-callout: none
+  -webkit-user-select: none
+  -khtml-user-select: none
+  -moz-user-select: none
+  -ms-user-select: none
+  user-select: none
```
Comment (peter-mouland) [inline src/toolkit.sass]: is this needed?
We have auto-prefixer in our webpack so expected this to be done automatically for us..

File: src/components/Accordion/AccordionItem.react.js [hunk 2]
```diff
@@ -10,8 +10,8 @@ class AccordionItem extends React.Component {
     const classes = classNames('now-accordion-item', [this.props.classes]);
 
     return (
-      <div className={classes} onClick={this.props.handleVisiblity}>
-        <span className="accordion-item-title">{this.props.title}</span>
+      <div className={classes}>
+        <a href="#" className="accordion-item-title" data-tracking-context={this.props.isOpen ? 'closing' : 'opening'} onClick={this.props.handleVisiblity}>{this.props.title}</a>
```
Comment (peter-mouland) [inline src/components/Accordion/AccordionItem.react.js]: I don't think we should force `data-tracking-*` into any react components. These should be optionally provided as props when using the component and passed through to the `a` tag in this case. i.e. via `{...this.props}` You could then write a test to ensure `data-trackin-*` is

Comment (peter-mouland) [inline src/components/Accordion/AccordionItem.react.js]: I don't think we should force `data-tracking-*` into any react components. This should be provided to the component as a prop, and only added onto the `a` tag if provided.

**Comment by peter-mouland** [comment]
This raises the question on how to handle tracking within distributable components.

I think the components should emit events (i.e. `open` and `close`) which the consuming app would listen for.  The consuming app would then emit the tracking event as required. 

This has the benefit of not tying in our toolkit components into a specific tracking library or a specific version of sky-tags.

File: src/components/Accordion/AccordionItem.react.js (hunk 2 above)
Comment (johnsibly) [inline src/components/Accordion/AccordionItem.react.js]: @acnalesso: I think it's the AccordionItem's job to set the context as this cannot be done externally as we have no access to the state of the accordion. The only bit that's hardcoded here is the context. The Omniture adapter in SkyTags does a great job by getting other values on the fly such as the link text. I don't think we want to be passing the same value in all our apps if it should be the same.

**Comment by johnsibly** [comment]
@acnalesso: @peter-mouland We could go as far as implementing it but the aim here is to get everyone to use SkyTags and if there is a new Third-party Analytics to be added it should be done in SkyTags as an adapter just like Omniture is. We are also not tied to a specific version of SkyTags as this is the generic attributes it looks for. These data-tracking* attributes have been used since sky.com.

I think by making the components emit events would just make it more complex for not much of a benefit as this is such a small think.

File: src/toolkit.sass (hunk 1 above)
Comment (johnsibly) [inline src/toolkit.sass]: @acnalesso: Sorry my bad. We will remove it then :)

Thanks

File: src/components/Accordion/AccordionItem.react.js (hunk 2 above)
Comment (alexflorisca) [inline src/components/Accordion/AccordionItem.react.js]: If you are to add tracking on the accordion, you should give the consuming app a choice of weather to track or not. So either having something like 'track=true' attribute or just passing the `data-track-context` prop through and setting it inside the component like you are at the moment. i.e only including `data-track-context` if the user actually wants to enable tracking.

**Comment by peter-mouland** [comment]
components should not be bound to each other i.e. the ract-toolkit should not depend on sky-tags.

If you bind them as suggested in this PR teams may have problems in the future. e.g. say you want a new button, but in that button has code for sky-tags v3.  This means instead of simply updating a small button component, you will also be forced to update your whole app to a new sky-tags which has a breaking change in it

**Comment by peter-mouland** [comment]
lets raise how to handle tracking as a new issue, and for now this is good to go
:+1:

**Comment by bentidy** [comment]
from @bdo :ok:

**Pull request merged**